kpi/renderers.py

```diff
@@ -122,6 +124,54 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
         )
 
 
+class SubmissionRendererExportBase(renderers.BaseRenderer, ExportBase):
+    def get_export_object(
+        self, renderer_context: Dict
+    ) -> Tuple[formpack.reporting.Export, Generator]:
+        view = renderer_context['view']
+        self.user = view.request.user
+        return super().get_export_object(
+            source=view.asset,
+            _async=False,
+        )
+
+
+class SubmissionXLSXRenderer(SubmissionRendererExportBase):
```

Comment:
I'm wondering whether `SubmissionRendererExportBase` is really needed?
We save 2 lines of code (4 if we count `SubmissionCSVRenderer`). 

`SubmissionXLSXRenderer` and `SubmissionCSVRenderer` could inherit directly from `renderers.BaseRenderer` and `ExportBase` (aka `ExportObjectMixin`)  and call 

```python
view = renderer_context['view']
self.user = view.request.user
```
just before `self.get_export_object()`

What do you think?

Reply:
Good idea 👍 I'll make that change.